### PR TITLE
feat: different table states

### PR DESCRIPTION
**Pull request by kayleegeorge**
add error table state and no token table state

Screen Shot 2022-07-21 at 3 38 40 PM

Screen Shot 2022-07-21 at 3 32 51 PM

### CODE REVIEW

File: src/components/Explore/TokenTable.tsx [hunk 1]
```diff
@@ -53,20 +79,30 @@ export default function TokenTable() {
...
         <HeaderRow timeframe={timePeriod} />
-        <NoTokenDisplay>No Favorited Tokens</NoTokenDisplay>
+        <NoTokenDisplay>You have no favorited tokens</NoTokenDisplay>
       </GridContainer>
     )
   }
 
   const topTokenAddresses = Object.keys(data)
   const showTokens = showFavorites ? favoriteTokens : topTokenAddresses
+  const filteredTokens = useFilteredTokens(showTokens)
```
Comment (kayleegeorge) [inline src/components/Explore/TokenTable.tsx:104]: @vm conditionally called here because we want to exit earlier if the data is null

Comment (vm) [reply src/components/Explore/TokenTable.tsx:104]: you need to put this at the top before you early exit, and have that hook return `[]`

Comment (vm) [reply src/components/Explore/TokenTable.tsx:104]: you also shouldn't pass `showTokens` in, call that hook inside `useFilteredTokens`

Comment (kayleegeorge) [reply src/components/Explore/TokenTable.tsx:104]: ay yai captain

File: src/components/Explore/TokenTable.tsx [hunk 2]
```diff
@@ -38,6 +44,26 @@ const LOADING_ROWS = Array(10)
     return <LoadingRow key={`${index}`} />
   })
 
+function useFilteredTokens(shownTokens: string[]) {
+  const filterString = useAtomValue(filterStringAtom)
+  const allTokens = useAllTokens()
+  const filteredTokens = shownTokens.filter((tokenAddress) => {
```
Comment (vm) [inline src/components/Explore/TokenTable.tsx:50]:
```
return useMemo(() => {
   ... all the `filteredTokens` logic here
}, [filterString, allTokens]
```

File: src/components/Explore/TokenTable.tsx [hunk 3]
```diff
@@ -38,11 +44,46 @@ const LOADING_ROWS = Array(10)
     return <LoadingRow key={`${index}`} />
   })
 
+function useFilteredTokens(shownTokens: string[]) {
+  const filterString = useAtomValue(filterStringAtom)
+  const allTokens = useAllTokens()
+
+  return useMemo(() => {
+    const filteredTokens = shownTokens.filter((tokenAddress) => {
```
Comment (vm) [inline src/components/Explore/TokenTable.tsx:52]: use `allTokens` here i think?

File: src/components/Explore/TokenTable.tsx [hunk 4]
```diff
@@ -38,11 +44,47 @@ const LOADING_ROWS = Array(10)
...
+      const addressIncludesFilterString = tokenAddress.toLowerCase().includes(lowercaseFilterString)
+      const nameIncludesFilterString = tokenName.toLowerCase().includes(lowercaseFilterString)
+      const symbolIncludesFilterString = tokenSymbol.toLowerCase().includes(lowercaseFilterString)
+      return nameIncludesFilterString || symbolIncludesFilterString || addressIncludesFilterString
+    })
+    return filteredTokens
+  }, [allTokens, shownTokens, filterString])
+}
+
+function NoTokensState(message: ReactNode, timePeriod: TimePeriod) {
```
Comment (vm) [inline src/components/Explore/TokenTable.tsx:73]: this should be a component

File: src/hooks/useTopTokens.ts [hunk 5]
```diff
@@ -283,7 +283,7 @@ const FAKE_TOP_TOKENS_RESULT = {
 }
 
 const useTopTokens = (): UseTopTokensResult => {
-  const [data, setData] = useState<TokenData | null>(null)
+  const [data, setData] = useState<TokenData>({})
```
Comment (vm) [inline src/hooks/useTopTokens.ts:286]: this seems non-ideal, why this change?

Comment (kayleegeorge) [reply src/hooks/useTopTokens.ts:286]: oh i thought you told me to do this LOL

Comment (kayleegeorge) [reply src/hooks/useTopTokens.ts:286]: but i understand now ok i will change it back

File: src/components/Explore/TokenTable.tsx (hunk 4 above)
Comment (kayleegeorge) [reply src/components/Explore/TokenTable.tsx:73]: but it uses all the same components as the table grid (since it still maintains the header and overall grid container). so I would need to export those styles which seems slightly redundant

Comment (kayleegeorge) [reply src/components/Explore/TokenTable.tsx:73]: like this is technical the table component, just a different token display (instead of token rows, it shows none)

Comment (vm) [reply src/components/Explore/TokenTable.tsx:73]: I just mean you should do `NoTokensState message={} timeperiod={}`

Comment (kayleegeorge) [reply src/components/Explore/TokenTable.tsx:73]: OH meme my b

**Review by vm — APPROVED** [review]
